esp32_lyrat/esp32_lyrat_idf4.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+#include "esp_err.h"
+#include "bsp/esp32_lyrat.h"
+#include "bsp_err_check.h"
+#include "esp_codec_dev_defaults.h"
+
+static const audio_codec_data_if_t *i2s_data_if = NULL;  /* Codec data interface */
+
+esp_err_t bsp_audio_init(const i2s_config_t *i2s_config)
+{
+    /* Setup I2S peripheral */
+    const i2s_pin_config_t i2s_pin_config = {
+        .mck_io_num = BSP_I2S_MCLK,
+        .bck_io_num = BSP_I2S_SCLK,
+        .ws_io_num = BSP_I2S_LCLK,
+        .data_out_num = BSP_I2S_DOUT,
+        .data_in_num = BSP_I2S_DSIN
+    };
+
+    /* Setup I2S channels */
+    const i2s_config_t std_cfg_default = BSP_I2S_DUPLEX_MONO_CFG_V4(22050);
+    const i2s_config_t *p_i2s_cfg = &std_cfg_default;
+    if (i2s_config != NULL) {
+        p_i2s_cfg = i2s_config;
+    }
+
+    ESP_ERROR_CHECK(i2s_driver_install(CONFIG_BSP_I2S_NUM, p_i2s_cfg, 0, NULL));
+    ESP_ERROR_CHECK(i2s_set_pin(CONFIG_BSP_I2S_NUM, &i2s_pin_config));
+
+    audio_codec_i2s_cfg_t i2s_cfg = {
+        .port = CONFIG_BSP_I2S_NUM,
+    };
+    i2s_data_if = audio_codec_new_i2s_data(&i2s_cfg);
+    BSP_NULL_CHECK(i2s_data_if, NULL);
+
+    /* Setup power amplifier pin */
+    const gpio_config_t io_conf = {
+        .intr_type = GPIO_INTR_DISABLE,
+        .mode = GPIO_MODE_OUTPUT,
+        .pin_bit_mask = BIT64(BSP_POWER_AMP_IO),
+        .pull_down_en = GPIO_PULLDOWN_DISABLE,
+        .pull_up_en = GPIO_PULLDOWN_DISABLE,
+    };
+    ESP_ERROR_CHECK(gpio_config(&io_conf));
```

Comment:
The PA pin should now be managed by esp_codec_dev. See https://github.com/espressif/esp-bsp/blob/31184767fc82edcd40ce05a7ed26b6390f94d30f/esp32_lyrat/esp32_lyrat.c#L188

So all API that init/deinit/control this pin should be removed